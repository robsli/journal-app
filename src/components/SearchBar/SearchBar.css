.search-container {
  align-items: flex-end;
  display: flex;
  justify-content: center;
  margin: 24px 0;
  padding: 0 8px;
}
.search-container > input {
  border: none;
  border-bottom: 1px solid gray;
  font-size: 1rem;
  font-weight: 600;
  height: 16px;
  padding: 8px 4px;
  width: 90%;
}
.search-container > input:focus {
  outline: none;
}
.search-container > .fa {
  color: gray;
  visibility: hidden;
  margin-left: -16px;
  transition: all 0.1s ease;
  -webkit-text-stroke: 1px white;
}
.search-container > i:hover {
  color: black;
}
.search-container > .clear-search {
  color: gray;
  padding-bottom: 6px;
  transform: rotate(90deg);
  visibility: visible;
}
.search-container > input:focus::-webkit-input-placeholder { color:transparent; }
.search-container > input:focus:-moz-placeholder { color:transparent; }