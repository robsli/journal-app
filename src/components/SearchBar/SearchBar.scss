.search-container {
  align-items: flex-end;
  display: flex;
  justify-content: center;
  margin: 24px 0;
  padding: 0 8px;

  input {
    border: none;
    border-bottom: 1px solid gray;
    font-size: 1rem;
    font-weight: 600;
    height: 16px;
    padding: 8px 4px;
    width: 90%;
  }
  input:focus {
    outline: none;
  }
  input:focus::-webkit-input-placeholder { color:transparent; }
  input:focus:-moz-placeholder { color:transparent; }

  .fa {
    background-color: white;
    color: gray;
    margin-left: -16px;
    margin-bottom: 8px;
    transition: all 0.1s ease;
    -webkit-text-stroke: 1px white;
  }

  i:hover {
    color: black;
    cursor: pointer;
  }
}
