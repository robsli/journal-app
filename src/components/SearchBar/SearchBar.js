import React from 'react'
import './SearchBar.css'

function SearchBar({
  handleChange,
  handleClearSearch,
  searchQuery
}) {
  return (
    <div className='search-container'>
      <input 
        type='text' 
        name='searchQuery' 
        value={ searchQuery } 
        onChange={ handleChange }
        placeholder='Enter search item here'
        size='10' />
      { searchQuery.length > 0 &&
        <i name='clearSearch' className='fa fa-times' aria-hidden='true' onClick={ handleClearSearch }></i>
      }
    </div>
  )
}

export default SearchBar
