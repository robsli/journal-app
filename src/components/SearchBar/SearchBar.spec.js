import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() })

import SearchBar from './SearchBar'

describe('SearchBar', () => {

  it('renders the default view correctly', () => {
    // Act
    const searchBar = renderer
      .create(<SearchBar searchQuery="" />)
      .toJSON()

    //Assert
    expect(searchBar).toMatchSnapshot()
  })

  it('renders the the user input correctly and displays clear input icon', () => {
    // Arrange
    const searchQuery = 'test'

    // Act
    const searchBar = renderer
      .create(<SearchBar searchQuery={searchQuery} />)
      .toJSON()
    
    // Assert
    expect(searchBar).toMatchSnapshot()
  })

  it('correctly calls the function to clear the search input field when the clear input icon is clicked', () => {
    // Arrange
    const searchQuery = 'test'
    const handleClearSearch = jest.fn()
    const wrapper = shallow(
      <SearchBar
        handleClearSearch={handleClearSearch}
        searchQuery={searchQuery}
      />
    )
    expect(handleClearSearch).not.toBeCalled()

    // Act
    wrapper.find('i').simulate('click')

    // Assert
    expect(handleClearSearch).toBeCalled()

  })

  it('clear input field icon disappears when searchQuery prop is cleared', () => {
    // Arrange
    const searchQuery = 'test'
    const wrapper = shallow(
      <SearchBar
        searchQuery={searchQuery}
      />
    )
    expect(wrapper.find('input').props().value).toEqual(searchQuery)

    // Act
    wrapper.setProps({ searchQuery: '' })
    const rendered = renderer.create(wrapper)

    // Assert
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
