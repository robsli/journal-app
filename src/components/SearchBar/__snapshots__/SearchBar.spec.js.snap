// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar clear input field icon disappears when searchQuery prop is cleared 1`] = `
<div
  className="search-container"
>
  <input
    name="searchQuery"
    onChange={undefined}
    placeholder="Enter search item here"
    size="10"
    type="text"
    value=""
  />
</div>
`;

exports[`SearchBar renders the default view correctly 1`] = `
<div
  className="search-container"
>
  <input
    name="searchQuery"
    onChange={undefined}
    placeholder="Enter search item here"
    size="10"
    type="text"
    value=""
  />
</div>
`;

exports[`SearchBar renders the the user input correctly and displays clear input icon 1`] = `
<div
  className="search-container"
>
  <input
    name="searchQuery"
    onChange={undefined}
    placeholder="Enter search item here"
    size="10"
    type="text"
    value="test"
  />
  <i
    aria-hidden="true"
    className="fa fa-times"
    name="clearSearch"
    onClick={undefined}
  />
</div>
`;
